005-structs-and-interfaces/structs.go

```diff
@@ -45,9 +64,28 @@ func (n *Barker) MakeNoise() string {
 	return "BARK BARK!!!!"
 }
 
+type GoodDog struct {
+	barker    Barker
+	isGoodDog bool
+}
+
+func (gd *GoodDog) MakeNoise() string {
```

Comment:
Good solution.

I will just add that there is a second solution using struct "embedding" which is similar to OOP inheritance.